RequestTypeTable

chengcxi opened the pull request:
closes #2 

Implemented RequestTypeTable and related test cases. 

Added delete & edit buttons for RequestTypes

- Students & users not logged in cannot edit /delete, only view 
- Professors and admins are able to edit/delete RequestTypes

chengcxi committed:
RequestTypeTable

chengcxi commented:
https://rec-chengcxi-dev.dokku-13.cs.ucsb.edu/

timothytwu commented:
I reviewed the components on storybook and it looks like everyone (user is now default role) has access to the edit and delete buttons. I checked the conditionals in your frontend/src/main/components/RequestType/RequestTypeTable.js file and I think you should double check the role restrictions again. I believe it should be profs/admins only can delete and edit

pconrad committed:
Merge branch 'main' into chengcxi-dev-requestTypeTable

chengcxi committed:
fixed user access permissions (now only professors can delete/edit)

chengcxi committed:
fixed user access permissions (now only professors can delete/edit)

chengcxi committed:
test

chengcxi committed:
test

chengcxi commented:
@timothytwu 

I have updated permissions, now only professors should be allowed to access the delete & edit buttons. Thank you for checking!

chengcxi committed:
Removed unrelated comments from code

Co-authored-by: Phill Conrad <pconrad@cs.ucsb.edu>

pconrad committed:
Merge branch 'main' into chengcxi-dev-requestTypeTable

pconrad commented:
So, the "see previous PR for details" is *maybe* acceptable in some circumstances, but:
* I'd like to see at least a little more context for *what* this PR is changing and *why*, even if the *how* is deferred to another PR description.
* At the very least, link to the specific PR description; it's as simple as putting a hashtag # followed by the PR number, and the link happens automatically.

chengcxi committed:
Delete frontend/src/fixtures/requestFixtures.js

chengcxi committed:
Delete frontend/src/stories/components/RequestType/RequestTypeTable.stories.js

chengcxi committed:
fix

chengcxi committed:
format

chengcxi committed:
test

chengcxi committed:
reformat

chengcxi committed:
rwar

chengcxi committed:
table

chengcxi committed:
tables

chengcxi committed:
stryker

chengcxi commented:
Updated PR description to be more useful
Updated RequestTableForms to be consistent 
Fixed permission issues. Now only professors are allowed access to edit/delete buttons

timothytwu commented:
@chengcxi checked the components on storybook and they all look good. Just wanted to make sure that you only wanted the professors to have edit and delete access. On the issue, it's a bit unclear but it might? also want admin access to edit and delete. But other than that it looks good

pconrad commented on frontend/src/main/components/RequestType/RequestTypeTable.js:
```diff
@@ -0,0 +1,69 @@
+    },
+  ];
+
+  // only professors can delete / edit
+  if (hasRole(currentUser, "ROLE_PROFESSOR")) {
+    columns.push(
+      ButtonColumn("Delete", "danger", deleteCallback, "RequestTypeTable"),
+    );
+  }
+
+  if (hasRole(currentUser, "ROLE_PROFESSOR")) {
+    columns.push(
+      ButtonColumn("Edit", "primary", editCallback, "RequestTypeTable"),
+    );
+  }
```
I think that ROLE_ADMIN should be able to do these things also, yes?

chengcxi replied:
Ah yes, I was a bit confused by the issue description. I'll add it right now

pconrad reviewed (changes requested):
Thanks ... this is much more clear!

But, I still wonder: shouldn't admins be able to edit and delete recommendation types also?

Can you make that change and update your PR description?

pconrad committed:
Merge branch 'main' into chengcxi-dev-requestTypeTable

pconrad committed:
Merge branch 'main' into chengcxi-dev-requestTypeTable

chengcxi committed:
added permissions for admins as well

chengcxi commented:
Updated permissions

Now admins have same access to create/delete button as professors.
Users and guests that are not logged in still have same access rights as before.

timothytwu reviewed (approved):
Checked the table component on storybook and verified that both admin and prof have edit and delete access for request types. Looks good to me

pconrad committed:
Merge branch 'main' into chengcxi-dev-requestTypeTable

pconrad reviewed (approved):
LGTM
